### ISSUE TITLE
Tackle fails to create cluster: "One of [--zone, --region] must be supplied"

### ISSUE THREAD

**Issue by cjwagner** [kubernetes/test-infra#10687]
**What happened**:
I ran Tackle to create a new cluster and deploy prow and got a cluster creation error after choosing a cluster name.
```sh
Get credentials for existing cluster or [create new]: 
Cluster name [prow]: prow
ERROR: (gcloud.container.clusters.create) One of [--zone, --region] must be supplied: Please specify location..
FATA[0059] Failed to select context                      error=create context: create cluster in colew-test: create cluster: exit status 1

```
**How to reproduce it (as minimally and precisely as possible)**:
`bazel run //prow/cmd/tackle`

**Anything else we need to know?**:
I think this is because my project does not specify a default region or zone. https://cloud.google.com/compute/docs/regions-zones/changing-default-zone-region

Tackle looks awesome!

/area prow/tackle
/assign @fejta

**Comment by idealhack** [comment]
I was unable to create a new cluster using Tackle and went with the manual way... but I don't remember the error message because I'm not sure which side it's on, Tackle or gcloud.

**Comment by grahamwhaley** [comment]
I'm having a similar issue. I want to deploy and try out `prow` in my local cluster, and don't have `gcloud` installed, so instantly run into:

```
 tackle
Existing kubernetes contexts:
* 0: kubernetes-admin@kubernetes (current)

Choose context or [create new]:
Getting active GCP account...WARN[0011] Run gcloud auth login to initialize gcloud
INFO[0011] Run gcloud auth login to initialize gcloud
FATA[0011] Failed to select context                      error=create context: get current project: exec: "gcloud": executable file not found in $PATH
```

Nothing I type at the `Choose context` prompt gets me around that.

The [docs hint I can run this on any k8s cluster](https://github.com/kubernetes/test-infra/blob/master/prow/getting_started_deploy.md#deploying-prow), and don't say I need `gcloud` :-). I'm guessing this is just a minor code issue, where it tries to log you in or scan `gcloud`, but in this instance the results would not be used anyway.

**Comment by grahamwhaley** [comment]
Adding to the above - when I do choose context '0', I get:

```
 tackle
Existing kubernetes contexts:
* 0: kubernetes-admin@kubernetes (current)

Choose context or [create new]: 0
Applying admin role bindings (to create RBAC rules)...
FATA[0001] Failed to apply cluster role binding to kubernetes-admin@kubernetes  error=current account: exec: "gcloud": executable file not found in $PATH
```

Maybe I have a local setup issue, but I'll dig into the code a bit, as the mention of `gcloud` in the error still feels wrong.

**Comment by grahamwhaley** [comment]
OK, looks like `tackle` is pretty hard wired to `gcloud` - I'll go with the [manual deployment](https://github.com/kubernetes/test-infra/blob/master/prow/getting_started_deploy.md#manual-deployment) like @idealhack did. Sorry for the noise ;-)

**Comment by idealhack** [comment]
@grahamwhaley It's OK. Thanks for trying out and feedbacks! We still have some hiccups here with tackle and gcloud 😅

**Comment by fejta** [comment]
/help

Fix would be to add a check for `gcloud config get-value compute/zone` and if it is empty either ask the user and/or tell them to run `gcloud config set compute/zone whatever`

https://github.com/kubernetes/test-infra/blob/accc8109e9529acaa064ae7df74cfe12d6cd0970/prow/cmd/tackle/main.go#L77-L79

**Comment by vishleshpatel** [comment]
Hi @fejta, I would like to pick this one.

**Comment by spiffxp** [comment]
/close
Closed by #13417
